### PR TITLE
gRPC 1.43 upgrade

### PR DESCRIPTION
**Pull request by veblush**
Backport #684 along with gRPC 1.43 upgrade.

### CODE REVIEW

**Comment by veblush** [comment]
@mprashanthsagar PTAL

File: pom.xml [hunk 1]
```diff
@@ -98,8 +98,8 @@
     <google.guava.version>31.0.1-jre</google.guava.version>
     <google.http-client.version>1.40.1</google.http-client.version>
     <google.oauth-client.version>1.32.1</google.oauth-client.version>
-    <google.protobuf.version>3.18.2</google.protobuf.version>
-    <grpc.version>1.41.1</grpc.version>
+    <google.protobuf.version>3.19.2</google.protobuf.version>
+    <grpc.version>1.43.1</grpc.version>
```
Comment (mprashanthsagar) [inline pom.xml:102]: If we are ok to upgrade these dependencies, can we backport #670 instead ? @medb

Comment (veblush) [reply pom.xml:102]: I like to minimize changes on dependencies because it could be a blocker for dataflow to get the latest 2.2.x patch.

**Comment by veblush** [comment]
I updated this PR to upgrade gRPC 1.43 only since #684 was already backported. Note that this change is required to use the Traffic Director properly.

**Review by mprashanthsagar — APPROVED** [review]

**Pull request merged**